packages/subcommands/src/lib/utils/SubCommandsHandler.ts

```diff
@@ -0,0 +1,152 @@
+			if (!subCommandInfo) break;
+
+			if (!subCommandInfo.command && !subCommandInfo.method) {
+				throw new SubCommandMissingOptionsError(subCommandInfo.name, parentName);
+			}
+
+			const soughtCommand = store.get(subCommandInfo.command ?? '') ?? (subCommandInfo.method ? command : undefined);
+			if (!soughtCommand) break;
+
+			parentName = soughtCommand.name;
+			iterableCommand = soughtCommand;
+			method = subCommandInfo.method;
+			args.save();
+
+			if (command === iterableCommand && subCommandInfo.method) break;
+		} while ((nameFromArgs = await args.pickResult('string')));
+
+		args.restore();
+		return { command: command === iterableCommand ? undefined : iterableCommand, parentName, method, args };
+	}
+
+	/**
+	 * Builds parent-child sub commands tree and make preconditions for them.
+	 * @param commands The commands store.
+	 */
+	public inspectSubCommandsTree(commands: CommandStore) {
+		for (const [, command] of commands.entries()) {
+			if (!command.subCommands) continue;
+
+			for (const { command: subCommandPieceName } of command.subCommands) {
+				if (!subCommandPieceName) continue;
+
+				const subCommand = commands.get(subCommandPieceName);
+				if (!subCommand) throw new SubCommandNotLoadedError(subCommandPieceName, command.name);
+				subCommand.isSubcommand = true;
+
+				// Setting up parent's preconditions to found sub command with filtering
+				if ((command.preconditions as PreconditionContainerArray).entries) {
+					const subCommandPreconditions = subCommand.preconditions as PreconditionContainerArray;
+					let parentPreconditionFiltered: IPreconditionContainer | null = command.preconditions;
+					for (const preconditionName of this.options.overlappedPreconditions) {
+						const a = (this.preconditionsDeepFilter(subCommandPreconditions, preconditionName) ??
+							new PreconditionContainerArray()) as PreconditionContainerArray;
+						if (subCommandPreconditions.entries.every((e, i) => e === a.entries[i])) continue;
+
+						parentPreconditionFiltered = this.preconditionsDeepFilter(parentPreconditionFiltered, preconditionName);
+					}
+
+					parentPreconditionFiltered && subCommandPreconditions.add(parentPreconditionFiltered);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Deep [[IPreconditionContainer]] filtering by container name. Will return null if result is empty.
+	 * @param container Some precondition container.
+	 * @param filterTarget The container name by which will be filtered the original one.
+	 */
+	private preconditionsDeepFilter(container: IPreconditionContainer | null, filterTarget: string) {
+		if (container instanceof PreconditionContainerArray) {
+			let containerArray = null;
+			for (const entry of container.entries) {
+				const result = this.preconditionsDeepFilter(entry, filterTarget);
+
+				if (!result) continue;
+				!containerArray && (containerArray = new PreconditionContainerArray());
+
+				containerArray?.add(result);
```

Comment:
You'll have one here
```suggestion
				containerArray.add(result);
```